build.gradle

```diff
@@ -143,6 +143,9 @@ ext {
     libs.junitPlatformLanucher,
     project(":test-common:test-common-runtime")
   ]
+
+  // See README.md for this build option details and example of usage
+  checkstyleVersion = project.hasProperty('checkstyleVersion') ? checkstyleVersion : versions.checkstyle
```

Comment:
Maybe we should configure the version in [dependencies.gradle](https://github.com/apache/kafka/blob/trunk/gradle/dependencies.gradle#L25), similar to how the Scala version is configured? This would ensure consistency and better version management.

Reply:
Hi @chia7712 and thanks for stopping by ! Please allow me to add some more details below.

_**Prologue**_ (relevant comment by @romani from CheksTyle team):  https://github.com/apache/kafka/pull/10656#issuecomment-835809154

This (now fairly dusty) PR was created in order to provide CheckStyle team with some handy way to include Kafka project into CheckStyle regression suite. As time went by PR slipped through the cracks (but I decided to stick with it and hence we are having this discussion now :smiley:).

So, if this PR gets into Kafka trunk CheckStyle team will be able to perform following sequence on their end, i.e. on their CI server (note: **_X.yz_** stands for any specific CheckStyle version):
```
git clone 
git checkout trunk
./gradlew check -PcheckstyleVersion=X.yz
```

As for a place where this change should reside ([dependencies.gradle](https://github.com/apache/kafka/blob/trunk/gradle/dependencies.gradle#L25) vs. [common build options](https://github.com/apache/kafka/blob/trunk/README.md#common-build-options)) I am open for a discussion.
Current proposal is aligned with suggestion made by @ijuma here: https://github.com/apache/kafka/pull/10656#issuecomment-836074067

Reply:
is there a way to run only chckstyle task ?
I think that "check" will run all validations.

What version of checkstyle is used now?

Reply:
> Current proposal is aligned with suggestion made by @ijuma here: https://github.com/apache/kafka/pull/10656#issuecomment-836074067

yes, it is good to use `./gradlew check -PcheckstyleVersion=X.yz` but it seems @ijuma 's comment is not related to the implementation. Personally, `versions.checkstyle` is the version used by whole project, and hence I prefer to handle the custom version in the  [dependencies.gradle](https://github.com/apache/kafka/blob/trunk/gradle/dependencies.gradle#L25)

Reply:
Hi @romani ! 

Currently Kafka is using Checkstyle version **8.36.2**, but today @m1a2st created this (pretty impressive) PR that bumps Checkstyle to **10.20.1**: https://github.com/apache/kafka/pull/17999/files#diff-02b139e755ab0e27e0b0a6f9845843ef189116955cc340c49a4022587dbb2b52 


As for a **_checkstyle-only_** Gradle task: I was under [impression](https://docs.gradle.org/8.10.2/userguide/checkstyle_plugin.html#dependencies_added_to_other_tasks) that `./gradlew check` is performing only Checkstyle checks, but after some experimenting I see that it not the case (maybe I can deal with that in a separate PR).

Reply:
> Personally, `versions.checkstyle` is the version used by whole project, and hence I prefer to handle the custom version in the [dependencies.gradle](https://github.com/apache/kafka/blob/trunk/gradle/dependencies.gradle#L25)

@chia7712 makes sense, `dependencies.gradle` it is then.

Reply:
@chia7712 pushed, let me know what you think (eventually I will squash commits into one, of course).

@romani we can continue our discussion related to fine-grained Checkstyle task(s); I will make a mental note to my self to open some tiket or draft some push request.

Reply:
Sure @romani, got that.

@chia7712 solution is squashed into one commit.